owtf/api/handlers/auth.py

```diff
@@ -0,0 +1,230 @@
+                "password": "Test@34335",
+                "confirm_password": "Test@34335",
+                "name": "test"
+            }
+
+            **Example response**:
+
+            .. sourcecode:: http
+
+            **Successful registration response**
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "ok"
+                }
+            }
+
+            **Failed registration response**
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "fail",
+                    "message": "Email already exists"
+                }
+            }
+
+        """
+        body_data = json.loads(self.request.body.decode('utf-8'))
+        name = body_data['name']
+        email = body_data['email']
+        password = body_data['password']
+        confirm_password = body_data['confirm_password']
+
+        already_taken = User.find_by_email(self.session, email)
+        reg = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&]{8,20}$"
+        pat = re.compile(reg)
+        mat = re.search(pat, password)
+        data = {
+            "status": "fail",
+        }
+        if password != confirm_password:
+            data["message"] = "Password doesn't match"
+            self.success(data)
+        elif not mat:
+            data["message"] = "Choose a strong password"
+            self.success(data)
+        elif already_taken:
+            data["message"] = "Email already exists"
+            self.success(data)
+        else:
+            hashed_pass = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt(8))
+            user = {}
+            user['email'] = email
+            user['password'] = hashed_pass
+            user['name'] = name
+            User.add_user(self.session, user)
+            data = {
+                "status": "ok",
+            }
+            self.success(data)
+
+
+class LogOutHandler(APIRequestHandler):
+    """Logs out the current user and clears the cookie"""
+
+    def get(self):
+        """
+            **Example request**:
+
+            .. sourcecode:: http
+
+            GET /api/v1/logout/ HTTP/1.1
+
+            **Example response**:
+
+            .. sourcecode:: http
+
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "ok"
+                }
+            }
+
+        """
+        self.clear_cookie("user")
```

Comment:
if we do not manage cookies, the only thing we need to do on a log out is invalidate the JWT of the user - its a much simpler state management than cookies.

Reply:
Added UserLoginToken model for storing the JWT token of the user which gets revoked when the user logs out.